src/many-identity-hsm/BUILD.bazel

```diff
@@ -0,0 +1,50 @@
+load("@crate_index//:defs.bzl", "aliases", "all_crate_deps")
+load("@rules_rust//rust:defs.bzl", "rust_library", "rust_test")
+
+package(default_visibility = [
+    "//src/many:__pkg__",
+])
+
+rust_library(
+    name = "many-identity-hsm",
+    srcs = glob(include = ["src/**/*.rs"]),
+    aliases = aliases(),
+    proc_macro_deps = all_crate_deps(
+        proc_macro = True,
+    ),
+    deps = all_crate_deps(
+        normal = True,
+    ) + [
+        "//src/many-error",
+        "//src/many-identity",
+        "//src/many-identity-dsa",
+        "//src/many-protocol",
+    ],
+)
+
+rust_library(
+    name = "many-identity-hsm-for-test",
+    srcs = glob(include = ["src/**/*.rs"]),
+    aliases = aliases(),
+    crate_name = "many_identity_hsm",
+    proc_macro_deps = all_crate_deps(
+        proc_macro = True,
+        proc_macro_dev = True,
+    ),
+    deps = all_crate_deps(
+        normal = True,
+        normal_dev = True,
+    ) + [
+        "//src/many-error",
+        "//src/many-identity:many-identity-for-test",
+        "//src/many-identity-dsa:many-identity-dsa-for-test",
+        "//src/many-protocol:many-protocol-for-test",
+    ],
+)
+
+rust_test(
```

Comment:
Is this working on your machine?

Reply:
<img width="2150" alt="image" src="https://user-images.githubusercontent.com/681969/187789329-3186da91-0479-41e6-9b29-2178df0a559e.png">

Reply:
Seems `mock` is flaky, not failing. I got it to fail once above, but now it passes.

For HSM, I get:

```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //src/many-identity-hsm:many-identity-hsm-test
-----------------------------------------------------------------------------

running 2 tests
test tests::hsm_ecdsa_sign_verify ... FAILED
test tests::hsm_ecdsa_sign_p256_verify ... FAILED

failures:

---- tests::hsm_ecdsa_sign_verify stdout ----
thread 'tests::hsm_ecdsa_sign_verify' panicked at 'Unable to load PKCS#11 module: LibraryLoading(DlOpen { desc: "dlopen(/usr/lib/softhsm/libsofthsm2.so, 0x0005): tried: \'/usr/lib/softhsm/libsofthsm2.so\' (no such file)" })', src/many-identity-hsm/src/lib.rs:229:50
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tests::hsm_ecdsa_sign_p256_verify stdout ----
Error: ManyError(Reason { code: HSMMutexPoisoned, message: Some("PKCS#11 global instance mutex poisoned:\n{details}"), arguments: {"details": "poisoned lock: another task failed inside"} })
thread 'tests::hsm_ecdsa_sign_p256_verify' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/test/src/lib.rs:185:5


failures:
    tests::hsm_ecdsa_sign_p256_verify
    tests::hsm_ecdsa_sign_verify

test result: FAILED. 0 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.02s
```

Reply:
Alright `bazel` needs the `--test_env`. I should have thought about it.

Still no idea why the first mock test failed. Can't repro even when I purge.